### PR TITLE
18 api documentation setup

### PR DESCRIPTION
**Pull request by hubertborysowski**
Please check if information provided in CONTRIBUTING.md is clear, understandable and sufficient so you'll be able to create documentation while creating your methods.

### CODE REVIEW

File: doc/CONTRIBUTING.md [hunk 1]
```diff
@@ -14,3 +14,14 @@ Suggested branches naming convetion: `task_number-task_name`.
 Commit message naming convention: `task_number if implemented this commit will`.
 
 To get task numbers, use: [Trello Card Numbers Plus](https://chrome.google.com/webstore/detail/trello-card-numbers-plus/ncibjlmfhjcjnphnpphgphbflpdpliei/related)
+
+## Documentation
```
Comment (Mateusz-Dobrzynski) [inline doc/CONTRIBUTING.md]: `API Documentation` would be more descriptive, as there are two types of documentation: the API documentation and documentation of the source code.

Comment (Mateusz-Dobrzynski) [reply doc/CONTRIBUTING.md]: Consider dividing this section of the documentation into docs' usage and creation (i.e. reading and writing).

File: doc/CONTRIBUTING.md [hunk 2]
```diff
@@ -14,3 +14,14 @@ Suggested branches naming convetion: `task_number-task_name`.
...
+
+The Swagger (OpenAPI 3) is used for API's documentation.
```
Comment (Mateusz-Dobrzynski) [inline doc/CONTRIBUTING.md]: Explain in more detail what is documented (the requests), what should be included in a given documentation entry, who should create the documentation (person developing/refactoring given request) and where should it be written (in a yaml file).

File: doc/CONTRIBUTING.md [hunk 3]
```diff
@@ -14,3 +14,14 @@ Suggested branches naming convetion: `task_number-task_name`.
...
+The documentation can be found under `http://localhost:8080/swagger-ui/index.html` path by default.
+Note that API must be online if you are willing to check the documentation.
+
+Information about initializing Swagger can be found under [this link](https://springdoc.org).
+Information about usage and examples of it can be found under these links:
+1. [OpenAPI 3, documentation overview](https://www.youtube.com/watch?v=2o_3hjUPAfQ)
```
Comment (Mateusz-Dobrzynski) [inline doc/CONTRIBUTING.md]: The documentation alone should be informative enough to allow contributing without additional research, especially watching time-consuming YouTube videos. Therefore I suggest removing this link.

File: doc/CONTRIBUTING.md [hunk 4]
```diff
@@ -14,3 +14,14 @@ Suggested branches naming convetion: `task_number-task_name`.
...
+2. [OpenAPI 3, basics](https://support.smartbear.com/swaggerhub/docs/tutorials/openapi-3-tutorial.html)
```
Comment (Mateusz-Dobrzynski) [inline doc/CONTRIBUTING.md]: Consider turning this block into a new section, e.g. _Further readings_.

**Review by Mateusz-Dobrzynski — COMMENTED** [review]
When reading this section of the documentation, the reader should know just enough to start documenting their code. What you have written is not enough though, as I don't even know which file should I edit to append to the documentation. Please include more information and address other remarks made in the comments.

**Review by Mateusz-Dobrzynski — CHANGES_REQUESTED** [review]
Refer to [this review](https://github.com/Mateusz-Dobrzynski/meals-management-system/pull/10#pullrequestreview-1589902509).

File: doc/CONTRIBUTING.md [hunk 5]
```diff
@@ -14,3 +14,105 @@ Suggested branches naming convetion: `task_number-task_name`.
...
+    @Operation(summary = "Get user by ID")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "User found"),
+        @ApiResponse(responseCode = "404", description = "User not found")
+    })
+    public ResponseEntity<User> getUserById(@PathVariable Long id) {
+        // Implementation
+    }
+}
+```
```
Comment (Mateusz-Dobrzynski) [inline doc/CONTRIBUTING.md]: This looks like an example copied from some external docs. If that's the case, remove this example and provide a link instead.

File: doc/CONTRIBUTING.md [hunk 6]
```diff
@@ -14,3 +14,105 @@ Suggested branches naming convetion: `task_number-task_name`.
...
+     * Handles /register requests by checking validity of provided data, registering
+     * new users in the database and sending confirmation emails.
+     * 
+     * @param user User object defined in a request body
+     * @return JSON response with registration token
+     * @throws Exception
+     */
+    @PostMapping(value = "/register", consumes = "application/json", produces = "application/json")
+    public ResponseEntity<String> registerUser(@RequestBody User user) throws Exception {...}
+```
```
Comment (Mateusz-Dobrzynski) [inline doc/CONTRIBUTING.md]: Using our project as an example is overall a good idea. Consider removing some lines for brevity (e.g. a Javadoc entry can be replaced with `//Javadoc entry`). Consider similar with imports, locals, etc. Also provide a source, i.e. link to the presented file **in its current state** (if you refer to the file in general, not a specific commit in particular, the example may soon become outdated).

File: doc/CONTRIBUTING.md [hunk 7]
```diff
@@ -14,3 +14,105 @@ Suggested branches naming convetion: `task_number-task_name`.
...
+
+    @Bean
+    public GroupedOpenApi customOpenApi() {
+        return GroupedOpenApi.builder()
+            .group("custom-api") // Define a group name for your APIs
+            .packagesToScan("com.example.controller") // Package(s) containing your API controllers
+            .build();
+    }
+}
+```
```
Comment (Mateusz-Dobrzynski) [inline doc/CONTRIBUTING.md]: Same for supposedly copy-pasted examples.

**Review by Mateusz-Dobrzynski — CHANGES_REQUESTED** [review]
Your documentation is now way more descriptive. However, I'm lacking any links to OpenAPI documentation which you replaced in favor of seemingly copy-pasted examples. Please remove these examples and provide some links as reference materials / further readings.

**Comment by hubertborysowski** [comment]
@Mateusz-Dobrzynski, well, I wasn't able to find a good example of basic OpenAPI's usage so I generated these examples with AI. Thus, I can't remove these examples unless I find some documentation that will meet this criteria, however I can add a few links to websites that expand the topic of OpenAPI.

**Comment by Mateusz-Dobrzynski** [comment]
@hubertborysowski, the links you provided in the previous commit combined with an example from our project will be sufficient. You should beware including AI-generated code, unless you tested it and know that it works. I suggest removing these examples.

**Comment by hubertborysowski** [comment]
@Mateusz-Dobrzynski, I'm aware of that. By coincidence, I tested similar methods in VSC yesterday and that's why I knew these examples will work. Nevertheless I understand what you mean. I'll stick to what you said then.

File: doc/CONTRIBUTING.md [hunk 8]
```diff
@@ -14,3 +14,48 @@ Suggested branches naming convetion: `task_number-task_name`.
...
+
+### Reading an API documentation
+To access API documentation turn the services online and go to the following path: `http://localhost:8080/swagger-ui.html`.
+
+OpenAPI provides a feature which allows user to send POST, GET, DELETE and PUT methods and receive a response.
+To access this feature, simply find desired endpoint in the documentation and click on it.
+
+### Additional literature
+1. [OpenAPI 3, additional examples](https://asbnotebook.com/spring-boot-openapi-3-example/)
+2. [OpenAPI 3, documentation overview - an YouTube video](https://www.youtube.com/watch?v=2o_3hjUPAfQ)
```
Comment (Mateusz-Dobrzynski) [inline doc/CONTRIBUTING.md]: I uphold what I written about YouTube videos (time-consuming, provide links to documentation instead). Other than that, this section is alright.

**Review by Mateusz-Dobrzynski — APPROVED** [review]
I still consider links to YouTube videos out of place. Other than that, it's acceptable.

File: doc/CONTRIBUTING.md (hunk 8 above)
Comment (hubertborysowski) [reply doc/CONTRIBUTING.md]: Well, you said about links that I provided in the previous commit, so I thought 'well, okay'. I'll remove that then.

**Pull request merged**